src/api/app/controllers/webui/monitor_controller.rb

```diff
@@ -74,8 +74,13 @@ def events
 
   private
 
+  UPPER_RANGE_LIMIT = 24 * 365
+
   def status_history(key, range)
-    MonitorControllerService::StatusHistoryFetcher.new(key, range.to_i).call
+    user_range = [UPPER_RANGE_LIMIT, range.to_i].min
+    Rails.cache.fetch("#{key}-#{user_range}", expires_in: (user_range.to_i * 3600) / 150) do
```

Comment:
> (user_range.to_i * 3600) / 150) 

Can you remove the magic numbers?

Reply:
I don't know what the `150` means here...

Reply:
okay...